src/env.ts

```diff
@@ -3,16 +3,30 @@ import { useEffect, useLayoutEffect } from 'react'
 export const IS_SERVER =
   typeof window === 'undefined' ||
   // @ts-ignore
-  !!(typeof Deno !== 'undefined' && Deno && Deno.version && Deno.version.deno)
+  !!(typeof Deno !== 'undefined' && Deno.version && Deno.version.deno)
```

Comment:
```suggestion
import { isUndefined } from './libs/helper'
export const IS_SERVER =
  isUndefined(window) ||
  // @ts-ignore
  !!(!isUndefined(Deno) && Deno.version && Deno.version.deno)
```

Reply:
I don't think you can use an undeclared variable (like `window`) before checking it with `typeof`:

![image](https://user-images.githubusercontent.com/3676859/117049334-912ea000-ad46-11eb-8cfc-0011b6e8ec6b.png)